panther_analysis_tool/main.py

```diff
@@ -475,18 +474,22 @@ def confirm_analysis_exists(args: argparse.Namespace, analysis_id_list: list) ->
     json_response = api_response["body"]
 
     json_output = json.loads(json_response)
-
     if len(analysis_id_list) != json_output["paging"]["totalItems"]:
-        analysis_found = {}
+
+        analysis_response = {}
```

Comment:
Just a suggestion, feel free to ignore if you don't find it easier to maintain, I think you can achieve the same with sets:

```python3
existing_ids = set([detection["id"] for detection in json_output["detections"]])
given_ids = set(analysis_id_list)
diff = given_ids.difference(existing_ids)
if diff:
    for detection_id in diff:
         # log 
    return list(existing_ids.intersection(given_ids))
return analysis_id_list
```